src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocGenerator.java

```diff
@@ -53,21 +61,30 @@ public static void generateXdocContent() throws Exception {
         for (Path path : templatesFilePaths) {
             final String pathToFile = path.toString();
             final File inputFile = new File(pathToFile);
-            final File outputFile = new File(pathToFile.replace(".template", ""));
-
+            final File tempFile = File.createTempFile(pathToFile.replace(".template", ""), "");
```

Comment:
What happens if the temp file already exists?

Reply:
Temp file by this method is always unique, that is why it is so convenient :)

Reply:
`@TempDir` just uses `Files.createTempDirectory` underneath. Peeking through the code, while it will try again if a directory exists, it looks like it can get stuck in an inifinite loop if there are enough existing directories matching up a Long number.

Also, race conditions are possible, which means there is a chance it is not unique. https://rules.sonarsource.com/java/rspec-5445/

Reply:
If someone will be so lucky, ok , he will rerun build.